vortexor/src/load_balancer.rs

```diff
@@ -0,0 +1,206 @@
+                    .to_string();
+                let exit = exit.clone();
+                let websocket_url = websocket_url.clone();
+                let server_to_slot = server_to_slot.clone();
+                let slot_sender = slot_sender.clone();
+                let highest_slot = highest_slot.clone();
+
+                Builder::new()
+                    .name(format!("load_balancer_subscription_thread-{ws_url_no_token}"))
+                    .spawn(move || {
+                        while !exit.load(Ordering::Relaxed) {
+                            info!("running slot_subscribe() with url: {websocket_url}");
+                            let mut last_slot_update = Instant::now();
+
+                            match PubsubClient::slot_subscribe(&websocket_url) {
+                                Ok((_subscription, receiver)) => {
+                                    while !exit.load(Ordering::Relaxed) {
+                                        match receiver.recv_timeout(Duration::from_millis(100))
+                                        {
+                                            Ok(slot) => {
+                                                last_slot_update = Instant::now();
+
+                                                server_to_slot
+                                                    .insert(websocket_url.clone(), slot.slot);
+                                                datapoint_info!(
+                                                        "rpc_load_balancer-slot_count",
+                                                        "url" => ws_url_no_token,
+                                                        ("slot", slot.slot, i64)
+                                                );
+
+                                                {
+                                                    let old_slot = highest_slot.fetch_max(slot.slot, Ordering::Relaxed);
+                                                    if slot.slot > old_slot {
+                                                        if let Err(e) = slot_sender.send(slot.slot)
+                                                        {
+                                                            error!("error sending slot: {e}");
+                                                            break;
+                                                        }
+                                                    }
+                                                }
+                                            }
+                                            Err(RecvTimeoutError::Timeout) => {
+                                                // RPC servers occasionally stop sending slot updates and never recover.
+                                                // If enough time has passed, attempt to recover by forcing a new connection
+                                                if last_slot_update.elapsed() >= Self::DISCONNECT_WEBSOCKET_TIMEOUT
```

Comment:
Is 30 seconds the right timeout? If vortexor stops getting updates for 30s, will it function right?